starter-code/pizza.js

```diff
@@ -17,69 +17,177 @@ var state = {
+  document.querySelector(".crust").classList.toggle("crust-gluten-free");
+  renderButtons(".btn-crust", state.glutenFreeCrust);
 }
 
-function renderButtons() {
+function renderButtons(foodClass, state) {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  //document.querySelector(foodClass).classList.toggle("active");
+
+  if (state) {
+    document.querySelector(foodClass).classList.add("active");
+  } else {
+    document.querySelector(foodClass).classList.remove("active");
+  }
 }
 
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
+  basePrice = 10;
+
+  if (state.pepperonni) {
+    basePrice += ingredients.pepperonni.price;
+    document.querySelector(".price li").style.visibility = "visible";
+    document.querySelector(".price li").style.position = "relative";
+  } else {
+    document.querySelector(".price li").style.visibility = "hidden";
+    document.querySelector(".price li").style.position = "absolute";
+  }
+  if (state.mushrooms) {
+    basePrice += 1;
+    //   document.querySelector(".price ul:nth-child(2)").style.visibility =
+    //     "visible";
+    //   document.querySelector(".price ul:nth-child(2)").style.position =
+    //     "relative";
+    // } else {
+    //   document.querySelector(".price ul:nth-child(2)").style.visibility =
+    //     "hidden";
+    //   document.querySelector(".price ul:nth-child(2)").style.position =
+    //     "absolute";
+  }
+  if (state.greenPeppers) {
+    basePrice += 1;
+    //   document.querySelector(".price ul:nth-child(3)").style.visibility =
+    //     "visible";
+    //   document.querySelector(".price ul:nth-child(3)").style.position =
+    //     "relative";
+    // } else {
+    //   document.querySelector(".price ul:nth-child(3)").style.visibility =
+    //     "hidden";
+    //   document.querySelector(".price ul:nth-child(3)").style.position =
+    //     "absolute";
+  }
+
+  if (state.whiteSauce) {
+    basePrice += 3;
+    //   document.querySelector(".price ul:nth-child(4)").style.visibility =
+    //     "visible";
+    //   document.querySelector(".price ul:nth-child(4)").style.position =
+    //     "relative";
+    // } else {
+    //   document.querySelector(".price ul:nth-child(4)").style.visibility =
+    //     "hidden";
+    //   document.querySelector(".price ul:nth-child(4)").style.position =
+    //     "absolute";
+  }
+
+  if (state.glutenFreeCrust) {
+    basePrice += 5;
+    //   document.querySelector(".price ul:nth-child(5)").style.visibility =
+    //     "visible";
+    //   document.querySelector(".price ul:nth-child(5)").style.position =
+    //     "relative";
+    // } else {
+    //   document.querySelector(".price ul:nth-child(5)").style.visibility =
+    //     "hidden";
+    //   document.querySelector(".price ul:nth-child(5)").style.position =
+    //     "absolute";
+  }
+
+  document.querySelector(".price strong").innerText = `$${basePrice}`;
```

Comment:
An other way to do this would be to loop through your state's keys, then add the price from your ingredient list of objects if the state of this ingredient is true